helpdesk_mgmt_assign_method/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 - TODAY, Kaynnan Lemes <kaynnan.lemes@escodoo.com.br>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    @api.model
+    def default_get(self, fields_list):
+        """Assign user based on team when creating a new ticket."""
+        res = super().default_get(fields_list)
+        team_id = res.get("team_id")
+        if team_id:
+            team = self.env["helpdesk.ticket.team"].browse(team_id)
+            user_id = team.get_new_user().id if team.get_new_user() else False
+            if (not fields_list or "user_id" in fields_list) and "user_id" not in res:
+                res["user_id"] = user_id
```

Comment:
```suggestion
        team_id = res.get("team_id")
        if (not fields_list or "user_id" in fields_list) and "user_id" not in res and team_id:
            team = self.env["helpdesk.ticket.team"].browse(team_id)
            res["user_id"] = team.get_new_user().id
```

nitpicking / not blocking: just changing the order of things to avoid calling `team.get_new_user()` if not needed